test/ibmq/test_ibmq_backends.py

```diff
@@ -18,65 +18,71 @@
 from unittest import skip
 from qiskit import ClassicalRegister, QuantumCircuit, QuantumRegister
 from qiskit.providers.exceptions import QiskitBackendNotFoundError
-from qiskit.providers.ibmq import IBMQ
 from qiskit.qobj import QobjHeader
-from qiskit.test import QiskitTestCase, requires_qe_access, slow_test
+from qiskit.test import requires_qe_access, slow_test
 from qiskit.compiler import assemble, transpile
 
+from .ibmqprovidertestcase import IBMQProviderTestCase
 
-class TestIBMQBackends(QiskitTestCase):
+
+class TestIBMQBackends(IBMQProviderTestCase):
     """Tests for all the IBMQ backends."""
 
-    def setUp(self):
+    def setUp(self):   # pylint: disable=invalid-name
+        """Required method for testing"""
         super().setUp()
-
         qr = QuantumRegister(1)
         cr = ClassicalRegister(1)
         self.qc1 = QuantumCircuit(qr, cr, name='circuit0')
         self.qc1.h(qr[0])
         self.qc1.measure(qr, cr)
 
+    @classmethod
+    def setUpClass(cls):  # pylint: disable=invalid-name
+        """Required method for testing"""
+        super().setUpClass()
+
     @requires_qe_access
     def test_remote_backends_exist(self, qe_token, qe_url):
         """Test if there are remote backends."""
-        IBMQ.enable_account(qe_token, qe_url)
-        remotes = IBMQ.backends()
+        self.provider.enable_account(qe_token, qe_url)
```

Comment:
Seeing that _all_ the `test_xx` methods end up enabling an account, perhaps we could take advantage of the superclass mechanism for creating a provider? As in (quick code, not tested!):

```python
    @requires_qe_access
    def _get_provider(self, qe_token, qe_url):
        """Return an instance of a Provider."""
        provider = self.provider_cls()
        provider.enable_account(qe_token, qe_url)
        return provider
```

In turn, this would make the tests in this file a bit more friendly (ie. no decorator, and no enable_account):
```python
    def test_remote_backend_status(self):
        """Test backend_status."""
        for backend in self.provider.backends():
            _ = backend.status()
```

And still be able to get preserve the behaviour of skipping tests when no credentials are passed, etc.

Reply:
If I understand correctly, the problem is there's no provider class called out in the file (new name) test_ibmq_providers.py.

Anyway, I tried it and in pylint I get:
```
test/ibmq/test_ibmq_provider.py:47:4: W0221: Parameters differ from overridden '_get_provider' method (arguments-differ)
test/ibmq/test_ibmq_provider.py:49:19: E1102: self.provider_cls is not callable (not-callable)
```

Reply:
Sorry for not mentioning it explicitly  - for the previous snippet, it was assuming moving the:
```
provider_cls = IBMQProvider
```
from https://github.com/Qiskit/qiskit-ibmq-provider/pull/124/files#diff-6eb0def457591c481d36f78c805cc523R26 into this file.

We can include a pylint disable in this case, as it is under control and the decorator appends parameters that throw the linter off - you can disable them via:
```
# pylint: disable=arguments-differ,not-callable
```
(although probably the not-callable will not be needed after moving the line above into the file).

Reply:
Oops, missed that :) ... trying that ...

Reply:
Looks good.